lib/Sema/CSDiag.cpp

```diff
@@ -5372,6 +5382,136 @@ bool FailureDiagnosis::diagnoseCallContextualConversionErrors(
   return false;
 }
 
+// Offer a fix-it for using `subscript` instead of the operator
+// if the inputs are compatible.
+static void diagnoseSubscriptMisuse(ApplyExpr *callExpr,
```

Comment:
It would be great if this function, like the rest of `diagnose*` returned `true|false` to identify that it did diagnose something.

Reply:
OK, it is a convention, I suppose. It’s of no interest to visitApplyExpr though.

Reply:
Right, that's because visit* methods returns AST nodes (or modified ones) or `nullptr` to identify termination.

Reply:
I meant this one 
https://github.com/AnthonyLatsis/swift/blob/d2d122ce977c8607e7b3ce2b3c3984fc59fca634/lib/Sema/CSDiag.cpp#L5544-5568